src/Raven.Server/RavenServer.cs

```diff
@@ -439,6 +451,29 @@ private void StartMonitoringCpuCredits()
                         Logger.Operations("Unhandled exception occured during cpu credit monitoring", e);
                 }
 
+                try
+                {
+                    if (sw.Elapsed.TotalSeconds >= (int)Configuration.Server.CpuCreditsExecSyncInterval.AsTimeSpan.TotalSeconds)
+                    {
+                        if (string.IsNullOrEmpty(Configuration.Server.CpuCreditsExec) == false)
+                        {
+                            var response = GetCpuCreditsFromExec();
+
+                            CpuCreditsBalance.RemainingCpuCredits = response.Remaining;
+                            CpuCreditsBalance.LastSyncTime = response.Timestamp;
+                            Array.Clear(CpuCreditsBalance.History, 0, CpuCreditsBalance.History.Length);
+                            CpuCreditsBalance.HistoryCurrentIndex = 0;
+                        }
+
+                        sw.Restart();
```

Comment:
do this maybe in `finally` so if something goes wrong this will be reset and we will not execute checks too often?

Reply:
If it's in the finally we'll always restart it and not get a chance to reach 30 minutes.

Instead, I'm changing the order of execution so that restart will always come before the attempt to get credits so that if we have a failure the stopwatch will still be restarted.